docs/tutorials/plugin.rst

```diff
@@ -0,0 +1,261 @@
+
+Add the following content to the ``myplugin.py`` file that you created earlier::
+
+    from tutor import hooks
+
+    hooks.Filters.ENV_PATCHES.add_item(
+        (
+            "openedx-lms-common-settings",
+            "FEATURES['ALLOW_PUBLIC_ACCOUNT_CREATION'] = False"
+        )
+    )
+
+Let's go over these changes one by one::
+
+    from tutor import hooks
+
+This imports the ``hooks`` module from Tutor, which grants us access to ``hooks.Actions`` and ``hooks.Filters`` (among other things).
+
+::
+
+    hooks.Filters.ENV_PATCHES.add_item(
+        <name>,
+        <content>
+    )
+
+This means "add ``<content>`` to the ``{{ patch("<name>") }}`` statement, thanks to the  ENV_PATCHES filter". In our case, we want to modify the LMS settings, both in production and development. The right patch for that is :patch:`openedx-lms-common-settings`. We add one item, which is a single Python-formatted line of code::
+
+    "FEATURES['ALLOW_PUBLIC_ACCOUNT_CREATION'] = False"
+
+.. note:: Notice how "False" starts with a capital "F"? That's how booleans are created in Python.
+
+Now, re-render your environment with::
+
+    $ tutor config save
+
+You can check that the feature was added to your environment::
+
+    $ grep -r ALLOW_PUBLIC_ACCOUNT_CREATION "$(tutor config printroot)/env"
+    /home/yourusername/.local/share/tutor/env/apps/openedx/settings/lms/production.py:FEATURES['ALLOW_PUBLIC_ACCOUNT_CREATION'] = False
+    /home/yourusername/.local/share/tutor/env/apps/openedx/settings/lms/development.py:FEATURES['ALLOW_PUBLIC_ACCOUNT_CREATION'] = False
+
+Your new settings will be taken into account by restarting your platform::
+
+    $ tutor local restart
+
+Congratulations! You've created your first working plugin. As you can guess, you can add changes to other files by adding other similar patch statements to your plugin.
+
+Modifying configuration
+=======================
+
+In the previous section you've learned how to add custom content to the Tutor templates. Now we'll see how to modify the Tutor configuration. Configuration settings fall in three categories:
+
+1. "base" settings that need to be preserved in config.yml: for instance, password and secret keys.
+2. "default" settings that should only be stored in config.yml when they are modified by users. Most settings are in this category.
+3. "override" settings that modify the base Tutor configuration. These will be removed and restored to their default values when the plugin is disabled.
```

Comment:
I'm OK with all of your suggestions, except for the renaming of "base" settings. While I agree that "base" is too generic, "required" is not appropriate either: after all, all settings are "required" to render environment templates. We need to find a word that says "unique to each user and thus must be persisted to disk". Maybe: "unique"? Or "user"?

Reply:
That is fair! I like "unique" a lot. "user" is OK too, although I think it's slightly more ambiguous, since some software uses the term "user settings" just to refer to user overrides of the default settings.

Reply:
We're going for "unique" then :)